templates/repo/diff/box.tmpl

```diff
@@ -195,6 +197,33 @@
 											{{ctx.Locale.Tr "repo.diff.bin_not_shown"}}
 										{{end}}
 									</div>
+								{{else if and $file.SubmoduleInfo}}
+									<div class="tw-p-3">{{svg "octicon-file-submodule"}} {{$submodule := $file.SubmoduleInfo -}}
+										{{- $refURL := $submodule.RefURL $.Repository.FullName -}}
+										{{- $submoduleName := $file.Name -}}
+										{{- $refShortID := ShortSha $submodule.RefID -}}
+										{{- if $file.IsDeleted -}}
+											{{- ctx.Locale.Tr "repo.diff.submodule_deleted" $submoduleName $refShortID -}}
+										{{- else if $file.IsCreated -}}
+											{{- $refAddedSubModule := $submoduleName -}}
+											{{- $refAddedAt := $refShortID -}}
+											{{- if $refURL -}}
+												{{- $refAddedSubModule = HTMLFormat `<a href="%s">%s</a>` $refURL $submoduleName -}}
+												{{- $refAddedAt = HTMLFormat `<a href="%s/commit/%s">%s</a>` $refURL (PathEscape $submodule.RefID) $refShortID -}}
+											{{- end -}}
+											{{- ctx.Locale.Tr "repo.diff.submodule_added" $refAddedSubModule $refAddedAt -}}
+										{{- else -}}
+											{{- $refUpdatedSubModule := $submoduleName -}}
+											{{- $refUpdatedFrom := ShortSha $submodule.PreviousRefID -}}
+											{{- $refUpdatedTo := $refShortID -}}
+											{{- if $refURL -}}
+												{{- $refUpdatedSubModule = HTMLFormat `<a href="%s">%s</a>` $refURL $submoduleName -}}
+												{{- $refUpdatedFrom = HTMLFormat `<a href="%s/commit/%s">%s</a>` $refURL (PathEscape $submodule.PreviousRefID) (ShortSha $submodule.PreviousRefID) -}}
+												{{- $refUpdatedTo = HTMLFormat `<a href="%s/commit/%s">%s</a>` $refURL (PathEscape $submodule.RefID) (ShortSha $submodule.RefID) -}}
```

Comment:
I'm not sure how best to do this in the translation system, but using partial formattted HTML like the following works.

```suggestion
												{{- $refUpdatedFrom = HTMLFormat `<a href="%s/compare/%s...%s">%s` $refURL (PathEscape $submodule.PreviousRefID) (PathEscape $submodule.RefID) (ShortSha $submodule.PreviousRefID) -}}
												{{- $refUpdatedTo = HTMLFormat `%s</a>` (ShortSha $submodule.RefID) -}}

```

Reply:
I think we could construct the URL links in Go instead of tmpl, will try it.

Reply:
Now the code could be simplified to this:

![image](https://github.com/user-attachments/assets/05e3eef9-2bb3-4f9e-b2d1-f3a868e1bf74)

Reply:
One more thing: we should avoid putting translatable strings between HTML tags, it caused many corrupts in history.

So ideally, we should not add links to a phrase of a sentence, only add links to a complete word.